src/common/providers/tasks.ts

```diff
@@ -88,22 +88,72 @@ export interface TaskContext {
    * The result of decoding and verifying an ODIC token.
    */
   auth?: AuthData;
+
+  /**
+   * The name of the queue.
+   * Populated via the X-CloudTasks-QueueName header.
+   */
+  queueName: string;
+
+  /**
+   * The "short" name of the task, or, if no name was specified at creation, a unique
+   * system-generated id.
+   * This is the my-task-id value in the complete task name, ie, task_name =
+   * projects/my-project-id/locations/my-location/queues/my-queue-id/tasks/my-task-id.
+   * Populated via the X-CloudTasks-TaskName header.
+   */
+  id: string;
+
+  /**
+   * The number of times this task has been retried.
+   * For the first attempt, this value is 0. This number includes attempts where the task failed
+   * due to 5XX error codes and never reached the execution phase.
+   * Populated via the X-CloudTasks-TaskRetryCount header.
+   */
+  retryCount: number;
+
+  /**
+   * The total number of times that the task has received a response from the handler.
+   * Since Cloud Tasks deletes the task once a successful response has been received, all
+   * previous handler responses were failures. This number does not include failures due to 5XX
+   * error codes.
+   * Populated via the X-CluodTasks-TaskExecutionCount header.
+   */
+  executionCount: number;
+
+  /**
+   * The schedule time of the task, as an RFC 3339 string in UTC time zone
+   * Populated via the X-CloudTasks-TaskETA header, which uses seconds since January 1 1970.
+   */
+  scheduledTime: string;
+
+  /**
+   * The HTTP response code from the previous retry.
+   * Populated via the X-CloudTasks-TaskPreviousResponse header
```

Comment:
Would suggest putting the additional comments like this under `@remark` as done here https://github.com/firebase/firebase-functions/blob/master/src/common/providers/https.ts#L148

Or to put it as a single sentence. FYI everything under `@remark` tag won't show up in the reference docs.